ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/page/v1/page/body.html

```diff
@@ -0,0 +1,21 @@
+<!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2021 Adobe
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+
+<main>
+    <sly data-sly-use.templatedContainer="com.day.cq.wcm.foundation.TemplatedContainer"
+         data-sly-repeat.child="${templatedContainer.structureResources}"
+         data-sly-resource="${child.path @ resourceType=child.resourceType, decorationTagName='div'}"></sly>
```

Comment:
Hi @vdua , I had this discussion with @jalagari  and @rismehta . we are wrapping everything inside body tag with main tag. We recommend writing styles on the basis of BEM selectors. until someone is searching for body>div, it won't cause a problem.

Reply:
DOM structure is generally part of public contract and changing DOM hierarchy is backward incompatible.we can't stop anyone from using this dom hierarchy in the code.

Reply:
>DOM structure is generally part of public contract and changing DOM hierarchy is backward incompatible.we can't stop anyone from using this dom hierarchy in the code.

+1, also, we can't expect all customers to consistently adhere to best practices, so this change is not backward compatible.

Reply:
@nit23uec - For transforming HTML to MD main tag is required according to helix API

1. Dont we ask customer to use BEM selectors for styling? 
2. Can you please please the public contract created for DOM hierarchy? Does sites also does it?

As highlighted it will only break for customer who are direct child  selector i.e. body > div

Reply:
> Dont we ask customer to use BEM selectors for styling?

Yes, that's the suggested method, although one could also utilize HTML tags for styling.

> Can you please please the public contract created for DOM hierarchy? Does sites also does it?

Yes both form and sites enforce the same. You can check the component documentation of each component. For example, https://github.com/adobe/aem-core-forms-components/tree/master/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput

https://github.com/adobe/aem-core-wcm-components/tree/main/content/src/content/jcr_root/apps/core/wcm/components/button/v2/button

Reply:
@deepprakash345 - lets go with new version of page component as suggested by @rismehta

Reply:
done